code/datums/brain_damage/special.dm

```diff
@@ -305,3 +305,244 @@
+		if(P.owner == owner.real_name)
+			owner_pda = P
+			break
+	if(!owner_pda || owner_pda.toff)
+		testing("Owner has no linked PDA open to messages")
+		QDEL_NULL(src)
+		return
+
+	brain_cam = new(owner)
+	last_snap = world.time // give them 2 minutes without getting harassed by pix
+	capone_appear_delay = world.time + 180 SECONDS // give them 3 minutes before capone shows up in person
+	..()
+
+/datum/brain_trauma/special/photo_friend/on_lose()
+	QDEL_NULL(brain_cam)
+	QDEL_NULL(capone)
+	..()
+
+/datum/brain_trauma/special/photo_friend/proc/start_snap()
+	testing("Starting snap: delay [snap_delay]")
+	snapping = TRUE
+	saved_location = get_turf(owner)
+	addtimer(CALLBACK(src, .proc/send_snap), snap_delay)
+
+/datum/brain_trauma/special/photo_friend/proc/send_snap()
+	testing("Sending?")
+	last_snap = world.time
+	snapping = FALSE
+	snap_delay = max(min_snap_delay, snap_delay - snap_delta)
+
+	var/pic_size = rand(2, 4)
+	var/datum/picture/selfie = brain_cam.captureimage(target=saved_location, user=null, size_x=pic_size, size_y=pic_size)
+
+	var/datum/signal/subspace/messaging/pda/signal = new(owner, list(
+		"name" = "???",
+		"job" = "???",
+		"message" = " ",
+		"targets" = list("[owner_pda.owner] ([owner_pda.ownjob])"),
+		"photo" = selfie
+	))
+
+	signal.send_to_receivers()
+
+/datum/brain_trauma/special/photo_friend/proc/create_capone()
+	if(world.time < capone_appear_delay)
+		return
+
+	var/list/turfs = list()
+	for(var/obj/structure/window/W in orange(owner, 6))
+		var/w_dist = get_dist(owner, W)
+		for(var/turf/open/O in orange(W, 1))
+			if(O in turfs)
+				continue
+			if(O.density)
+				continue
+			if(get_dist(owner, O) <= w_dist)
+				continue
+			if(locate(/obj/structure/window) in O.contents)
+				continue
+			turfs += O
+
+	if(turfs.len == 0)
+		return 
+
+	var/turf/open/where = pick(turfs)
+	capone = new(where, owner)
+	capone.victim = owner
+
+/datum/brain_trauma/special/photo_friend/on_life()
+	//&& (world.time > last_snap + snap_delay) 
+	if(!snapping && prob(30))
+		QDEL_NULL(capone)
+		start_snap()
+
+	if(QDELETED(capone) || !capone.loc || capone.z != owner.z)
+		QDEL_NULL(capone)
+		create_capone()
+		return
+
+	if(get_dist(owner, capone) >= 9 && prob(10))
+		QDEL_NULL(capone)
+		create_capone()
+
+	if(get_dist(owner, capone) <= 3)
+		//2do: some other effect for this
+		QDEL_NULL(capone)
+
+	if(prob(5))
+		QDEL_NULL(capone)
+
+	..()
+
+/obj/effect/hallucination/simple/capone
+	name = "Your Best Friend"
+	desc = "Good to see you again!"
+	image_icon = 'icons/mob/human_face.dmi'
+	image_state = "lips_spray_face"
+	var/victim
+
+/obj/effect/hallucination/simple/capone/New()
+	START_PROCESSING(SSprocessing,src)
+	..()
+
+/obj/effect/hallucination/simple/capone/process()
+	setDir(get_dir(src, victim)) //doesnt actually do anything rn
+
+/obj/effect/hallucination/simple/capone/Destroy()
+	STOP_PROCESSING(SSprocessing,src)
+	return ..()
+
+
+
+
+
+/datum/brain_trauma/special/photo_friend/cute
```

Comment:
A lot of the procs look duplicated from the parent type. Parent procs gets called automatically from this subtype if none exists.

Reply:
Oh I just left those duplicated just in case I'd need them later, this isn't really review ready. I'm probably just gonna cut the cute version tonight anyway

Reply:
Aight, this wasnt a thorough review or anything. Let us know when its ready